### PR TITLE
add missing AutoGPU guards

### PR DESCRIPTION
**Pull request by soumith**

### CODE REVIEW

**Comment by ezyang** [comment]
LGTM. Do we have a test for the bug this fixes?

**Comment by soumith** [comment]
this doesn't fix any bugs that surfaced. but i found these spots while investigating another bug (the actual fix was #2743 ).

File: torch/csrc/autograd/functions/basic_ops.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@ auto Error::apply(const variable_list& grad_outputs) -> variable_list {
 auto DelayedError::apply(const variable_list& inputs) -> variable_list {
   tensor_list outputs;
   outputs.reserve(inputs.size());
+  AutoGPU guard(inputs[0]);
```
Comment (apaszke) [inline torch/csrc/autograd/functions/basic_ops.cpp]: I don't think this one is needed

File: torch/csrc/autograd/functions/tensor.cpp [hunk 2]
```diff
@@ -133,6 +133,7 @@ auto Cat::apply(const variable_list& inputs) -> variable_list {
 }
 
 auto Chunk::apply(const variable_list& inputs) -> variable_list {
+  AutoGPU guard(inputs[0].data());
```
Comment (apaszke) [inline torch/csrc/autograd/functions/tensor.cpp]: This shouldn't need AutoGPU, because there are no data allocations

**Comment by soumith** [comment]
made requested changes and merged into master

**Pull request closed without merging**